pennylane/transforms/core/transform.py

```diff
@@ -177,15 +178,45 @@ def qnode_circuit(a):
             )
 
     # 3: CHeck the classical co-transform
-    if classical_cotransform is not None:
-        if not callable(classical_cotransform):
-            raise TransformError("The classical co-transform must be a valid Python function.")
+    if classical_cotransform is not None and not callable(classical_cotransform):
+        raise TransformError("The classical co-transform must be a valid Python function.")
 
-    dispatcher = TransformDispatcher(
+    return TransformDispatcher(
         quantum_transform,
         expand_transform=expand_transform,
         classical_cotransform=classical_cotransform,
         is_informative=is_informative,
         final_transform=final_transform,
     )
-    return dispatcher
+
+
+def null_postprocessing(results):
+    """A postprocessing function with null behavior."""
+    return results[0]
+
+
+def expand_fn_transform(
+    expand_fn: Callable[["pennylane.tape.QuantumTape"], "pennylane.tape.QuantumTape"]
+) -> "TransformDispatcher":
+    """Construct a transform from a tape-> tape function.
+
+    Args:
+        expand_fn (Callable): a function from a single tape to a single tape
+
+    Returns:
+
+        .TransformDispatcher: Returns a transform dispatcher object that that can transform any
+        circuit-like object in PennyLane.
+
+    >>> device = qml.device('default.qubit.legacy', wires=2)
+    >>> my_transform = qml.transforms.core.expand_fn_transform(device.expand_fn)
+    >>> my_transform
+    <transform: expand_fn>
```

Comment:
To check I understand - the main utility of this new function is that you don't need to worry about the postprocessing function?